boa_engine/src/context/mod.rs

```diff
@@ -246,15 +257,134 @@ impl<'host> Context<'host> {
     /// Compile the script AST into a `CodeBlock` ready to be executed by the VM.
     pub fn compile_script(&mut self, statement_list: &StatementList) -> JsResult<Gc<CodeBlock>> {
         let _timer = Profiler::global().start_event("Script compilation", "Main");
+
+        let var_declarations = top_level_var_scoped_declarations(statement_list);
```

Comment:
Can you add a link to https://tc39.es/ecma262/#sec-globaldeclarationinstantiation? Just to know where this came from.